e2e/test/MessageSendFaultInjectionTests.cs

```diff
@@ -380,86 +338,1452 @@ public async Task Message_GracefulShutdownSendRecovery_MqttWs()
+                    Client.TransportType.Amqp_Tcp_Only,
+                    MuxWithPoolingPoolSize,
+                    MuxWithPoolingDevicesCount,
+                    ConnectionStringAuthScope.Device,
+                    FaultInjection.FaultType_Throttle,
+                    FaultInjection.FaultCloseReason_Boom,
+                    FaultInjection.DefaultDelayInSec,
+                    FaultInjection.DefaultDurationInSec).ConfigureAwait(false);
+        }
+
+        [Ignore]
+        [TestMethod]
+        [TestCategory("ConnectionPoolingE2ETests")]
+        // WIP: Exception handling
+        public async Task Message_DeviceSak_ThrottledConnectionRecovery_MuxWithPooling_AmqpWs()
+        {
+            await SendMessageRecoveryMuxedOverAmqp(
+                    TestDeviceType.Sasl,
+                    Client.TransportType.Amqp_WebSocket_Only,
+                    MuxWithPoolingPoolSize,
+                    MuxWithPoolingDevicesCount,
+                    ConnectionStringAuthScope.Device,
+                    FaultInjection.FaultType_Throttle,
+                    FaultInjection.FaultCloseReason_Boom,
+                    FaultInjection.DefaultDelayInSec,
+                    FaultInjection.DefaultDurationInSec).ConfigureAwait(false);
+        }
+
+        [Ignore]
+        [TestMethod]
+        [TestCategory("ConnectionPoolingE2ETests")]
+        // WIP: Exception handling
+        public async Task Message_IoTHubSak_ThrottledConnectionRecovery_MuxWithoutPooling_Amqp()
+        {
+            await SendMessageRecoveryMuxedOverAmqp(
+                    TestDeviceType.Sasl,
+                    Client.TransportType.Amqp_Tcp_Only,
+                    MuxWithoutPoolingPoolSize,
+                    MuxWithoutPoolingDevicesCount,
+                    ConnectionStringAuthScope.IoTHub,
+                    FaultInjection.FaultType_Throttle,
+                    FaultInjection.FaultCloseReason_Boom,
+                    FaultInjection.DefaultDelayInSec,
+                    FaultInjection.DefaultDurationInSec).ConfigureAwait(false);
+        }
+
+        [Ignore]
+        [TestMethod]
+        [TestCategory("ConnectionPoolingE2ETests")]
+        // WIP: Exception handling
+        public async Task Message_IoTHubSak_ThrottledConnectionRecovery_MuxWithoutPooling_AmqpWs()
+        {
+            await SendMessageRecoveryMuxedOverAmqp(
+                    TestDeviceType.Sasl,
+                    Client.TransportType.Amqp_WebSocket_Only,
+                    MuxWithoutPoolingPoolSize,
+                    MuxWithoutPoolingDevicesCount,
+                    ConnectionStringAuthScope.IoTHub,
+                    FaultInjection.FaultType_Throttle,
+                    FaultInjection.FaultCloseReason_Boom,
+                    FaultInjection.DefaultDelayInSec,
+                    FaultInjection.DefaultDurationInSec).ConfigureAwait(false);
+        }
+
+        [Ignore]
+        [TestMethod]
+        [TestCategory("ConnectionPoolingE2ETests")]
```

Comment:
Please separate the tests by file. In general, avoid using TestCategory on methods and use it per class.